### ISSUE TITLE
Parse Error: Class "PhantomCamera3D" hides a global script class.

### ISSUE THREAD

**Issue by TranquilMarmot** [ramokz/phantom-camera#275]
### Issue description

Godot v4.2.2.stable.mono - macOS 14.1.1 - Vulkan (Forward+) - integrated Apple M1 - Apple M1 (8 Threads)
Phantom Camera v0.7.0.4

On startup, Godot logs the following:
```
 res://addons/phantom_camera/scripts/phantom_camera/phantom_camera_3D.gd:3 - Parse Error: Class "PhantomCamera3D" hides a global script class.
  modules/gdscript/gdscript.cpp:2726 - Failed to load script "res://addons/phantom_camera/scripts/phantom_camera/phantom_camera_3D.gd" with error "Parse error". (User)
  res://addons/phantom_camera/scripts/resources/camera_3D_resource.gd:2 - Parse Error: Class "Camera3DResource" hides a global script class.
  modules/gdscript/gdscript.cpp:2726 - Failed to load script "res://addons/phantom_camera/scripts/resources/camera_3D_resource.gd" with error "Parse error". (User)
```

Likewise, when running the game it immediately crashes with:
```
Parser Error: Class "PhantomCamera3D" hides a global script class.
```

Somehow, this is broken even rolling back the files changed by the auto-update 😢 
To get the project running again, I had to delete the local `.godot` directory, re-load the project, and then reject the phantom camera update.

### Steps to reproduce

Upgraded from Phantom Camera v0.6 (I think? I had cloned the repo) to v0.7.0.4 via auto-update

**Comment by ramokz** [comment]
Dup of #263 and #272

tldr: this is more of a Godot editor issue than an addon one, where it randomly struggles with file renaming at times. Some report that reopening the project multiple times resolved their issues.
